custom/rch/views.py

```diff
@@ -0,0 +1,335 @@
+from custom.rch.forms import BeneficiariesFilterForm
+from custom.rch.models import RCHChildRecord, RCHMotherRecord, AreaMapping
+from custom.rch.templatetags.view_tags import mask_aadhar_number
+from custom.rch.utils import (
+    valid_aadhar_num_length,
+)
+from custom.rch.const import (
+    ICDS_CAS_DOMAIN,
+    RECORDS_PER_PAGE,
+    AADHAR_NUM_FIELDS,
+)
+
+
+@method_decorator([toggles.VIEW_CAS_RCH_REPORTS.required_decorator(),
+                   login_and_domain_required], name='dispatch')
+class BeneficariesList(JSONResponseMixin, BaseDomainView):
+    urlname = 'rch_cas_dashboard'
+    http_method_names = ['get', 'post']
+
+    # ToDo: Check how to set page title. Looks like this is not working
+    page_title = ugettext_noop("RCH-CAS Beneficiary list")
+    template_name = 'rch/beneficiaries_list.html'
+
+    def section_url(self):
+        return reverse(self.urlname, args=[ICDS_CAS_DOMAIN])
+
+    @method_decorator(require_superuser)
+    @use_select2
+    @use_angular_js
+    def dispatch(self, request, *args, **kwargs):
+        response = super(BeneficariesList, self).dispatch(request, *args, **kwargs)
+        # set cookie for pagination
+        response.set_cookie('hq.pagination.limit.rch_cas_beneficiaries.%s' % request.domain, RECORDS_PER_PAGE)
+        return response
+
+    def _set_rch_beneficiaries(self, include_matched_beneficiaries, only_matched=False, in_data=None):
+        """
+        Location Hierarchy: State -> District -> Village -> AWC
+        Filters down-up Village -> District -> State
+        Tries to filters by village_id selected but then falls back to
+        district's or state's village id for filtering
+        When filtering using AWC id, it looks for the corresponding village id using
+        AWC-Village mapping but it would result in beneficiaries from other AWC since
+        its many to many mapping.
+        :param include_matched_beneficiaries: include matched beneficiaries or
+         just render rch records that have no matching CAS record yet
+        :param only_matched: filter only matched beneficiaries
+        :return: True if it could find records else False
+        """
+        beneficiary_type = in_data.get('beneficiary_type')
+        if beneficiary_type == 'child':
+            self.beneficiaries = RCHChildRecord.objects
+        else:
+            self.beneficiaries = RCHMotherRecord.objects
+        if only_matched:
+            # exclude beneficiaries where cas_case_id is null
+            self.beneficiaries = self.beneficiaries.exclude(cas_case_id__isnull=True)
+        elif not include_matched_beneficiaries:
+            # fetch beneficiaries where case_case_id is NOT null
+            self.beneficiaries = self.beneficiaries.exclude(cas_case_id__isnull=False)
+        village_ids = []
+
+        village_code = in_data.get('village_code')
+        if village_code:
+            village_ids = village_ids + [village_code]
```

Comment:
`village_ids` will always be an empty list here so it doesn't seem useful to include it. Just have `village_ids = [village_code]`